src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -25,7 +25,7 @@
     }
 
     private final Supplier<RubyNode> resolver;
-    private RubyNode resolved = null;
+    private volatile RubyNode resolved = null;
```

Comment:
I cannot see what it fixes, the variable is only read/written in the atomic section.